executor/runtime/types/container_test.go

```diff
@@ -668,3 +668,41 @@ func TestServiceMeshEnabled(t *testing.T) {
 	assert.NotNil(t, svcMeshConf)
 	assert.Equal(t, svcMeshConf.Image, imgName)
 }
+
+func TestServiceMeshEnabledWithConfig(t *testing.T) {
+	// If service mesh is set to enabled, but neither the `ProxydServiceImage` config value
+	// or the passhtrough property are set, service mesh should end up disabled
+	config := config.Config{
+		ContainerServiceMeshEnabled: true,
```

Comment:
Can you add a negative/absent test, where we have an empty config and we assert that there is not a servicemesh sidecar?

Reply:
Yeah, I added that in the last commit.  It's part of `TestNewContainer()` above.